changelog.d/13106.bugfix

```diff
@@ -0,0 +1 @@
+Validates room alias before calling internal functions
```

Comment:
Here's my suggestion:

```suggestion
Fix a long-standing bug where room directory requests would cause an internal server error if given a malformed room alias.
```